### ISSUE TITLE
How to calculate the accuracy of the trained model?

### ISSUE THREAD

**Issue by talhakabakus** [AlexeyAB/darknet#3218]
Once the training is finished, what is your practice to calculate the accuracy of the trained model? I am aware of that we could use the confidences but (as you know) it does not guarantee that the confidences belong to the right detection. So, I think we could annotate the test images as we do for the training images, then use the annotations inside the `.txt` files to calculate if the detection is on the right object. How do you calculate the accuracy of your model? Is there a command/parameter for this?

**Comment by AlexeyAB** [comment]
Use this command to get accuracy (mAP): https://github.com/AlexeyAB/darknet#when-should-i-stop-training
`./darknet detector map data/obj.data yolo-obj.cfg backup\yolo-obj_9000.weights`

**Comment by talhakabakus** [comment]
That's what actually I have been looking for! Thanks @AlexeyAB for this great implementation. Last question, would it be enough to use the mAP value, which is 87.50% for my case (as the output of the result is added below), while evaluating the model?

```
 calculation mAP (mean average precision)...
20
 detections_count = 37, unique_truth_count = 18
class_id = 0, name = ball, ap = 87.50%           (TP = 17, FP = 6)

 for thresh = 0.25, precision = 0.74, recall = 0.94, F1-score = 0.83
 for thresh = 0.25, TP = 17, FP = 6, FN = 1, average IoU = 62.92 %

 IoU threshold = 50 %, used Area-Under-Curve for each unique Recall
 mean average precision (mAP@0.50) = 0.875039, or 87.50 %
Total Detection Time: 0.000000 Seconds
```

**Comment by saadyousuf45** [comment]
cannot get values of TP therefore map is 0% please help 
![image](https://user-images.githubusercontent.com/33414652/59153884-7561e700-8aa8-11e9-92d9-55dcb34b29ec.png)

**Comment by verepcode** [comment]
> Use this command to get accuracy (mAP): https://github.com/AlexeyAB/darknet#when-should-i-stop-training
> `./darknet detector map data/obj.data yolo-obj.cfg backup\yolo-obj_9000.weights`

@AlexeyAB , I have finished yolov3-tiny training with my own data and tried your suggested map command but got nothing on bash screen. Then I have tried with "recall" instead of "map", got  
Couldn't open file: data/coco_val_5k.list error
but I could not try to train or test with coco data. I have 
valid = data/ test_obj.txt  and  test_obj.txt in my darknet directory. I did not find any suggestion about this issue. I think, it is my local problem. Any suggestion is fair, thanks for your help.

**Comment by AlexeyAB** [comment]
@verepcode 

* Use this repository instead of https://github.com/pjreddie/darknet
* Look at your `data/obj.data` file and set `valid = data/test_obj.txt`

**Comment by verepcode** [comment]
@AlexeyAB You are right. I was training yolov3-tiny using pjreddie's repo.At now, I'm using your repo and trying to train the same model with same cfg settings but an error has occurred,
darknet: ./src/parser.c:315: parse_yolo: Assertion `l.outputs == params.inputs' failed.
Aborted (core dumped)
I have found that the filter settings are not suitable for the class number but as I said before, I'm using the same cfg file. I had changed the filters=255 to my settings, filters = 3*(numclasses + 5) so I have one class, filters=18.
When I was using pjreddie's repo, this error did not occurred.

**Comment by verepcode** [comment]
I have realized my mistake. It was just related to a missed class line in yolov3-tiny_obj.cfg so that the system could not understand which number of classes should be used for the filter.
